### ISSUE TITLE
[Bug]: unbound module failed to start

### ISSUE THREAD

**Issue by 3hedgehogs** [netdata/netdata#17280]
### Bug description

If module unbound is active but unbound service is not running the plugin is crashed

```
~/usr/libexec/netdata/plugins.d$ ./go.d.plugin -d -m unbound
DBG godplugin/main.go:145 plugin: name=go.d, version=1.45.1.0 component=agent
DBG godplugin/main.go:147 current user: name=netdata, uid=996 component=agent
INF godplugin/main.go:151 env HTTP_PROXY '', HTTPS_PROXY '' component=agent
INF agent/agent.go:145 instance is started component=agent
INF agent/setup.go:23 loading config file component=agent
DBG agent/setup.go:31 looking for 'go.d.conf' in [/opt/netdata/etc/netdata /opt/netdata/usr/lib/netdata/conf.d] component=agent
INF agent/setup.go:38 found '/opt/netdata/etc/netdata/go.d.conf component=agent
INF agent/setup.go:45 config successfully loaded component=agent
INF agent/agent.go:149 using config: enabled 'true', default_run 'true', max_procs '2' component=agent
INF agent/setup.go:50 loading modules component=agent
INF agent/setup.go:73 enabled/registered modules: 1/77 component=agent
INF agent/setup.go:79 building discovery config component=agent
DBG agent/setup.go:126 looking for 'unbound.conf' in [/opt/netdata/etc/netdata/go.d /opt/netdata/usr/lib/netdata/conf.d/go.d] component=agent
DBG agent/setup.go:142 found '/opt/netdata/etc/netdata/go.d/unbound.conf component=agent
INF agent/setup.go:147 dummy/read/watch paths: 0/1/0 component=agent
INF discovery/manager.go:116 registered discoverers: [file discovery: [file reader] service discovery] component="discovery manager"
DBG agent/setup.go:165 looking for 'vnodes/' in [/opt/netdata/etc/netdata /opt/netdata/usr/lib/netdata/conf.d] component=agent
INF agent/setup.go:177 found '/opt/netdata/usr/lib/netdata/conf.d/vnodes' (0 vhosts) component=agent
INF discovery/manager.go:61 instance is started component="discovery manager"
INF functions/manager.go:50 instance is started component="functions manager"
INF jobmgr/manager.go:87 instance is started component="job manager"
DBG functions/ext.go:15 registering function 'config' component="functions manager"
INF file/discovery.go:69 instance is started component=discovery discoverer=file
CONFIG go.d:collector:unbound create accepted template /collectors/jobs internal 'internal' 'add schema enable disable test' 0x0000 0x0000

INF sd/sd.go:66 instance is started component="service discovery"
INF file/read.go:48 instance is started component=discovery discoverer=file
INF file/read.go:49 instance is stopped component=discovery discoverer=file
DBG jobmgr/manager.go:122 received configs: 1/+1/-0 ('/opt/netdata/etc/netdata/go.d/unbound.conf') component="job manager"
CONFIG go.d:collector:unbound:local create accepted job /collectors/jobs user 'discoverer=file_reader,file=/opt/netdata/etc/netdata/go.d/unbound.conf' 'schema get enable disable update restart test' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating unbound[local] job, config: map[__provider__:file reader __source__:discoverer=file_reader,file=/opt/netdata/etc/netdata/go.d/unbound.conf __source_type__:user address:127.0.0.1:8953 autodetection_retry:20 cumulative_stats:false module:unbound name:local priority:70000 timeout:1 tls_cert:/etc/unbound/unbound_control.pem tls_key:/etc/unbound/unbound_control.key tls_skip_verify:true update_every:30 use_tls:true] component="job manager"
INF unbound/init.go:21 reading '/etc/unbound/unbound.conf' collector=unbound job=local
INF unbound/init.go:43 applying configuration: ["statistics-cumulative": '', "control-enable": 'yes', "control-interface": '127.0.0.1', "control-port": '8953', "control-user-cert": '', "control-key-file": '', "control-cert-file": ''] collector=unbound job=local
DBG unbound/unbound.go:85 using address: 127.0.0.1:8953, cumulative: false, use_tls: true, timeout: 1s collector=unbound job=local
DBG unbound/unbound.go:87 using tls_skip_verify: true, tls_key: /etc/unbound/unbound_control.key, tls_cert: /etc/unbound/unbound_control.pem collector=unbound job=local
ERR unbound/unbound.go:96 failed to connect: dial tcp 127.0.0.1:8953: connect: connection refused collector=unbound job=local
ERR module/job.go:238 check failed collector=unbound job=local
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x358 pc=0x70eb77]

```

### Expected behavior

Old versions

```
INF unbound/init.go:21 reading '/etc/unbound/unbound.conf' collector=unbound job=local
INF unbound/init.go:43 applying configuration: ["statistics-cumulative": '', "control-enable": 'yes', "control-interface": '127.0.0.1', "control-port": '8953', "control-user-cert": '', "control-key-file": '', "control-cert-file": ''] collector=unbound job=local
DBG unbound/unbound.go:85 using address: 127.0.0.1:8953, cumulative: false, use_tls: true, timeout: 1s collector=unbound job=local
DBG unbound/unbound.go:87 using tls_skip_verify: true, tls_key: /etc/unbound/unbound_control.key, tls_cert: /etc/unbound/unbound_control.pem collector=unbound job=local
ERR unbound/unbound.go:103 failed to connect: dial tcp 127.0.0.1:8953: connect: connection refused collector=unbound job=local
ERR module/job.go:229 check failed collector=unbound job=local
INF jobmgr/manager.go:225 unbound[local] job detection failed, will retry in 20 seconds component="job manager"
REPORT_JOB_STATUS 'unbound' 'local_5098870865835851024' 'error' 0 'job detection failed, will retry later'
```

### Steps to reproduce

1. Install/upgrade to 1.45.1
2. Stop unbound service
3. Configure unbound.conf in etc/netdata/go.d/
4. Activate unbound module in go.d.cof
5. Restart netdata

Or

"stop unbound"
cd usr/libexec/netdata/plugins.d/
sudo -u netdata -s
./go.d.plugin -d -m unbound

### Installation method

kickstart.sh

### System info

```shell
Linux hostname 6.5.11-6-pve #1 SMP PREEMPT_DYNAMIC PMX 6.5.11-6 (2023-11-29T08:32Z) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="12"
/etc/os-release:VERSION="12 (bookworm)"
/etc/os-release:VERSION_CODENAME=bookworm
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.45.1
    Installation Type __________________________________________ : manual-static
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /opt/netdata/etc/netdata
    Stock Configurations _______________________________________ : /opt/netdata/usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /opt/netdata/var/cache/netdata
    Permanent Databases ________________________________________ : /opt/netdata/var/lib/netdata
    Plugins ____________________________________________________ : /opt/netdata/usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /opt/netdata/usr/share/netdata/web
    Log Files __________________________________________________ : /opt/netdata/var/log/netdata
    Lock Files _________________________________________________ : /opt/netdata/var/lib/netdata/lock
    Home _______________________________________________________ : /opt/netdata/var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.5.11-6-pve
    Operating System ___________________________________________ : unknown
    Operating System ID ________________________________________ : unknown
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : unknown
    Operating System Version ID ________________________________ : 12
    Detection __________________________________________________ : unknown
Hardware:
    CPU Cores __________________________________________________ : 6
    CPU Frequency ______________________________________________ : 2262000000
    RAM Bytes __________________________________________________ : 10737418240
    Disk Capacity ______________________________________________ : 290984034304
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : lxc
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : Debian GNU/Linux
    Container Operating System ID ______________________________ : debian
    Container Operating System ID Like _________________________ : unknown
    Container Operating System Version _________________________ : 12 (bookworm)
    Container Operating System Version ID ______________________ : 12
    Container Operating System Detection _______________________ : /etc/os-release
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine ___________________________________________________ : YES
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : NO
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

### Additional info

It is inside Proxmox LXC container

**Comment by ilyam8** [comment]
Do you have the full panic stack, @3hedgehogs? I can't reproduce the issue.

**Comment by ilyam8** [comment]
Thanks for reporting, @3hedgehogs. Fixed in #17283.
